sync15-adapter/src/token.rs

```diff
@@ -167,30 +196,40 @@ enum TokenState {
 /// The TokenserverClient - long lived and fetches tokens on demand (eg, when
 /// first needed, or when an existing one expires.)
 #[derive(Debug)]
-pub struct TokenserverClient {
-    // The stuff needed to fetch a token.
-    base_url: String,
-    access_token: String,
-    key_id: String,
+struct TokenserverClientImpl<TF: TokenFetcher> {
```

Comment:
I've been told in the Rust IRC that it's considered bad style to put type constraints on structs directly, and instead you should just put them in the impls/functions/etc.

I don't really understand the logic for that, but given that putting it on the struct doesn't free you from putting them everywhere else, it's not like it matters.

You should be able to just delete the `: TokenFetcher` here.